lazyflow/classifiers/vigraRfAdaptiveMaskPixelwiseClassifier.py

```diff
@@ -0,0 +1,198 @@
+
+class VigraRfAdaptiveMaskPixelwiseClassifierFactory(LazyflowPixelwiseClassifierFactoryABC):
+    """
+    An implementation of LazyflowPixelwiseClassifierFactoryABC using a vigra RandomForest.
+    This exists for testing purposes only. (it is normally better to use the vector-wise 
+    classifier so lazyflow can cache the feature matrices).
+    This implementation is simple and un-optimized.
+    """
+    VERSION = 1 # This is used to determine compatibility of pickled classifier factories.
+                # You must bump this if any instance members are added/removed/renamed.
+                    
+    def __init__(self, *args, **kwargs):
+        self._args = args
+        self._kwargs = kwargs
+    
+    def create_and_train_pixelwise(self, feature_images, label_images, axistags=None):
+        logger.debug( 'training pixel-wise vigra RF' )
+        
+        all_features = numpy.ndarray( shape=(0, feature_images[0].shape[-1]), dtype=numpy.float32 )
+        all_labels = numpy.ndarray( shape=(0,1), dtype=numpy.uint32 )
+
+        # Extract label points and corresponding feature vectors
+        for feature_image, label_image in zip(feature_images, label_images):
+            label_coords = numpy.nonzero(label_image)[:-1] # discard channel
+            label_vector = label_image[label_coords]
+            feature_matrix = feature_image[label_coords]
+            all_features = numpy.concatenate((all_features, feature_matrix), axis=0)
+            all_labels = numpy.concatenate((all_labels, label_vector), axis=0)
+        
+        # Save for future reference
+        known_labels = numpy.unique(all_labels)
+
+        assert len(all_features) == len(all_labels)
+        classifier = vigra.learning.RandomForest(*self._args, **self._kwargs)
+        classifier.learnRF(all_features, all_labels)
+
+        return VigraRfAdaptiveMaskPixelwiseClassifier( classifier, known_labels )
+
+    def get_halo_shape(self, data_axes):
+        # No halo necessary, but since this classifier is for testing purposes, let's add one anyway.
+        halo = tuple(range( len(data_axes)-1 ))
+        return halo + (0,) # (no channel halo)
+
+    def estimated_ram_usage_per_requested_predictionchannel(self):
+        return 4
+
+    @property
+    def description(self):
+        temp_rf = vigra.learning.RandomForest( *self._args, **self._kwargs )
+        return "Vigra Random Forest ({} trees)".format( temp_rf.treeCount() )
+
+    def __eq__(self, other):
+        return (    isinstance(other, type(self))
+                and self._args == other._args
+                and self._kwargs == other._kwargs )
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+assert issubclass( VigraRfAdaptiveMaskPixelwiseClassifierFactory, LazyflowPixelwiseClassifierFactoryABC )
+
+class VigraRfAdaptiveMaskPixelwiseClassifier(LazyflowPixelwiseClassifierABC):    
+    """
+    Adapt the vigra RandomForest class to the interface lazyflow expects.
+    """
```

Comment:
This class needs a better docstring.  Since the idea here is to put this work and pick it up later, we'll need help remembering what this class does and what the next steps to take are.  Regarding the next steps, mention some TODO items, such as parallelization, profile/optimize the masking step (is there a faster option than scipy's `binary_dilation()` function?), and also mention the need to think about issues at the image borders.  (The current implementation works if you're predicting the entire frame, but for spatially-divided blocks, objects that "walk in" from the side will be missed.)